src/components/media-views.js

```diff
@@ -694,6 +694,20 @@ AFRAME.registerComponent("media-video", {
       if (url.startsWith("hubs://")) {
         const streamClientId = url.substring(7).split("/")[1]; // /clients/<client id>/video is only URL for now
         const stream = await NAF.connection.adapter.getMediaStream(streamClientId, "video");
+        // We subscribe to video stream notifications for this peer to update the video element
+        // This could happen in case there is an ICE failure that requires a transport recreation.
+        this._onStreamUpdated = async (peerId, kind) => {
```

Comment:
Don't think this will actually happen in practice since I don't think we ever change the `src` for a video running from a MediaStream, but it might be worth checking if `this._onStreamUpdated` is non null here and removing the listener before  registering the new one.

Reply:
Updated